implementation/building.tex

```diff
@@ -0,0 +1,165 @@
+\section{Build System}
+\label{sec:build_system}
+
+Building the library and the EFM32 executables turned out to be one of the major parts of the work done in this project.
+This section describes how building the project have changed over time.
+This includes managing dependencies, compiling the RCL and REL and the Silicon Labs EFM32 \texttt{emlib}.
+We have also utilized a continuous integration system that has helped us to keep the project up to date with the nightly builds of Rust and making sure that the builds have been consistent across the systems it has been built on.
+
+\subsection{Early days}
+\label{ssub:using_make}
+
+\todo{Maybe mention more about armboot in background instead of here.}
+When the project first started out it was based upon the \texttt{armboot} \cite{github:armboot} git repository available on GitHub.
+\texttt{armboot} is a small template project for using Rust bare metal on a STM32 ARM \glspl{mcu}.
+These are, similarly with the EFM32 series, also based on the Cortex-M series of ARM processor cores.
+
+\todo{Håvard: Skriv denne seksjonen}
+The project got up and running pretty fast based on how \texttt{armboot} was built.
+We looked at armboot's Makefile to figure out what flags to pass to \texttt{rustc} in order to make it cross-compile for an ARM target architecture.
+
+\subsection{Transitioning to Cargo}
+\label{ssub:transitioning_to_cargo}
+
+It was always a goal to use Cargo for building, distributing and managing the packages and dependencies that would become part of this project.
```

Comment:
Include a why for this: 
 - To lower the bar for a Rust programmer to make use of it.
 - To see/evaluate if it was possible